### PR TITLE
Restore buy screen from legacy app

### LINKED ISSUE #2369: [redeem] Restore BUY screen in dApp
### Requirements

- [x] restore the Buy Carbon form (as is) from https://legacy-app.klimadao.finance/#/redeem on https://app.klimadao.finance/#/redeem
- [x] show the Risk Disclaimer if user enters the site from /redeem URL

### PR DESCRIPTION
**Pull request by 0xMakka**
## Description

Restore the buy screen on the dApp.

## Related Ticket

Closes #2369

### CODE REVIEW

**Review by jabby09 — CHANGES_REQUESTED** [review]
@0xMakka getting an error on the final SUBMIT step.
APPROVE was successful.
Screenshot 2024-06-25 at 9 33 50 AM

**Comment by 0xMakka** [comment]
> @0xMakka getting an error on the final SUBMIT step. APPROVE was successful. Screenshot 2024-06-25 at 9 33 50 AM

Hmm @jabby09 , will take a look again. I successfully bought before submitting the PR earlier (paying with Klima).

**Comment by 0xMakka** [comment]
@jabby09, the first time it failed for me with the same error you faced. I tried changing my Polygon RPC on metamask and after this I tested paying with both Klima and USDC and both were successful. 

Screenshot 2024-06-25 at 13 45 15

Screenshot 2024-06-25 at 13 44 15

Klima - https://polygonscan.com/tx/0x68e67f3d6d266d9ddcfa4072c76ea68618f26f416b6abc697daa4bc28a1b0278
USDC - https://polygonscan.com/tx/0xc6bba929f2926c88f099d594f39620db25ee6f37c168723e156a9a5dd18cc337

**Review by jabby09 — COMMENTED** [review]
@0xMakka new error this morning on retest

I can't get any of the filters to work ??
Screenshot 2024-06-26 at 9 20 05 AM

**Review by jabby09 — APPROVED** [review]

File: lib/constants/index.ts [hunk 1]
```diff
@@ -278,8 +278,10 @@ export const offsetCompatibility: CompatMap = {
   fiat: ["bct", "nct", "mco2", "ubo", "nbo"],
 };
 
+// TODO - remove once env variable is added...
```
Comment (Atmosfearful) [inline lib/constants/index.ts:281]: still a TODO? env should be added.

But note that I am pushing for us to never use env vars in `lib` as it should be environment agnostic...

**Review by Atmosfearful — APPROVED** [review]
I did not QA myself but the code looks mergeable!
1 comment

File: lib/constants/index.ts (hunk 1 above)
Comment (0xMakka) [reply lib/constants/index.ts:281]: @Atmosfearful I originally had the subgraph urls in here and must have left this comment here. I've actually moved the new env variables and subgraph url for bridged carbon to `app/lib/constants.ts` as part of this PR as I remembered you saying we shouldn't be adding env variables to `lib`. Will remove this comment.

**Pull request merged**